node/webpack-quick-start/build.electron.webpack.ElectronWebpackQuickStart.yaml

```diff
@@ -1,84 +1,83 @@
 # Use the Electron 2 BaseApp, which adds several common libraries we'll need.
 base: org.electronjs.Electron2.BaseApp
-base-version: '19.08'
-# Add the Node 10 SDK extension.
+base-version: '21.08'
+# Add the Node 12 SDK extension.
 sdk-extensions:
-  - org.freedesktop.Sdk.Extension.node10
+  - org.freedesktop.Sdk.Extension.node12
 # Electron doesn't use a traditional locale format so separate-locales is useless.
 separate-locales: false
 command: start-electron-webpack-quick-start
 finish-args:
-  # These two lines add Xorg access for graphics.
-  - '--share=ipc'
-  - '--socket=x11'
+  # These three lines add the permissions needed for graphics.
+  - --device=dri
+  - --share=ipc
+  - --socket=x11
   # Sound access.
-  - '--socket=pulseaudio'
+  - --socket=pulseaudio
   # Network access.
-  - '--share=network'
+  - --share=network
   # If you need to access the filesystem, also add:
-  # - '--filesystem=home'
+  # - --filesystem=home
 modules:
   # With electron-webpack and electron-builder we don't install Node to /app/node,
   # because electron-builder will bundle everything for us in one piece.
   # Instead we jump straight to the quick start module.
 
-  # However, since this quick start uses yarn, we do have to install that.
-  - name: yarn
-    buildsystem: simple
-    build-commands:
-      - 'cp -a * /app'
-    # Only used for building, so clean it up afterwards.
-    cleanup:
-      - '*'
-    sources:
-      - type: archive
-        url: https://github.com/yarnpkg/yarn/releases/download/v1.19.1/yarn-v1.19.1.tar.gz
-        sha256: 34293da6266f2aae9690d59c2d764056053ff7eebc56b80b8df05010c3da9343
-
   - name: electron-webpack-quick-start
     buildsystem: simple
     build-options:
-      # Add the node bin directory & yarn directory.
-      append-path: '/usr/lib/sdk/node10/bin:/app/yarn/bin'
+      # Add the node bin directory.
+      append-path: /usr/lib/sdk/node12/bin
       env:
-        # Set the Electron cache directory.
-        # (The directory format is: /run/build/MODULE_NAME/flatpak-node/electron-cache)
-        ELECTRON_CACHE: '/run/build/electron-webpack-quick-start/flatpak-node/electron-cache'
-        # Sets the directory where Node is located so way npm won't download the headers.
-        npm_config_nodedir: '/usr/lib/sdk/node10'
+        # Set the cache directory, used to find Electron and some other helper
+        # tools.
+        # (The directory format is: /run/build/MODULE_NAME/flatpak-node/cache)
+        XDG_CACHE_HOME: /run/build/electron-webpack-quick-start/flatpak-node/cache
+        # Sets the directory where Node is located so that npm won't download the headers.
+        npm_config_nodedir: /usr/lib/sdk/node12
     build-commands:
       # Have Yarn use the offline mirror.
-      - 'HOME=$PWD yarn config --offline set yarn-offline-mirror $FLATPAK_BUILDER_BUILDDIR/flatpak-node/yarn-mirror'
+      - HOME=$PWD yarn config --offline set yarn-offline-mirror $FLATPAK_BUILDER_BUILDDIR/flatpak-node/yarn-mirror
```

Comment:
This is rather hackish, I'd rather suggest adding a separate `.yarnrc` source instead:
```yaml
sources:
  ...
  - type: inline
    contents: |
      yarn-offline-mirror /run/build/electron-webpack-quick-start/flatpak-node/yarn-mirror
    dest-filename: .yarnrc